### ISSUE TITLE
Documentation and skeleton file for elgg-plugin.php

### ISSUE THREAD

**Issue by cleverthings** [Elgg/Elgg#10987]
I think the code is complete for elgg-plugin.php, but I don't see the documentation for it.  In the file layout given in the Plugin Developer page it is listed, but I don't see an example file or description of what it does or how to use it:
http://learn.elgg.org/en/stable/guides/plugins/plugin-skeleton.html

**Comment by mrclay** [comment]
The features have the docs. E.g. [views](http://learn.elgg.org/en/master/guides/views.html#views-and-third-party-assets). I think that's probably where they should like, but I can see an argument for a page dedicated to the file, linking to those.

**Comment by mrclay** [comment]
Your perspective as a newer Elgg devs is highly needed here.

**Comment by cleverthings** [comment]
That views link was exactly what I was looking for - thanks! I was building the plugin from scratch and when I created that file (just an empty placeholder), it broke the plugin because it didn't return an array to core. BUT had I kept reading in the manual, in the chapter on 3rd party views (the link you posted) the answer was there waiting for me. The Notes and Warnings like those on the Views are especially helpful for new developers! Somehow y'all know exactly what destructive and ill-advised direction I was headed towards. ;)  I might add a hyperlink from that file listed on the skeleton page to that views page, but the docs are fine.

I think I see the advantages of handling views that way instead of a start file, and will be looking for more examples in the newer plugins.  I'll let you decide if it's worth creating more content in that area, but I'm good with closing this issue. The answer was there, I just didn't dig deep enough. It showed up as the 6th search result, but somehow I missed it. Sorry!
